specification/securitycenter/resource-manager/Microsoft.Security/preview/2015-06-01-preview/security.json

```diff
@@ -0,0 +1,1975 @@
+                            "type": "string",
+                            "description": "Short constant description of the error"
+                        },
+                        "message": {
+                            "type": "string",
+                            "description": "Detailed description of the error and resolution suggestions"
+                        }
+                    }
+                }
+            }
+        },
+        "Resource": {
+            "properties": {
+                "id": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource Id"
+                },
+                "name": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource name"
+                },
+                "type": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "Resource type"
+                }
+            },
+            "required": [
+                "id",
+                "name",
+                "type"
+            ],
+            "x-ms-azure-resource": true,
+            "description": "Describes an Azure resource."
+        },
+        "Kind": {
+            "properties": {
+                "kind": {
+                    "type": "string",
+                    "description": "Kind of the resource"
+                }
+            },
+            "required": [
+                "kind"
+            ],
+            "x-ms-azure-resource": true,
+            "description": "Describes an Azure resource with kind"
+        },
+        "Location": {
+            "properties": {
+                "location": {
+                    "type": "string",
+                    "description": "Location where the resource is stored"
+                }
+            },
+            "required": [
+                "location"
+            ],
+            "x-ms-azure-resource": true,
+            "description": "Describes an Azure resource with location"
+        },
+        "TaskList": {
+            "description": "List of security task recommendations",
+            "properties": {
+                "value": {
+                    "readOnly": true,
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Task"
+                    }
+                }
+            }
+        },
+        "Task": {
+            "description": "Security task that we recommend to do in order to strengthen security",
+            "properties": {
+                "properties": {
+                    "readOnly": true,
+                    "type": "object",
+                    "properties": {
+                        "state": {
+                            "readOnly": true,
+                            "type": "string",
+                            "enum": ["Active", "Dismissed", "Resolved", "InProgress"]

```

Comment:
Enums are good for validation and intellisense. If the service decides to add a new value to the enum in the future then it causes breaking changes in the sdk. Why have an enum when the property is readOnly (can only be sent by the server in the response)?

Reply:
removed the enum

Reply:
@chlahav - Make sure you make similar changes wherever possible. provisioningState is another candidate.